src/typings/errors.d.ts

```diff
@@ -0,0 +1,3 @@
+export interface Error {
```

Comment:
A type Error exist by default in ts, I don't think we should redefine it

Reply:
I will change the name, good catch.

Reply:
Why do you even need it?

Reply:
It's not absolutely necessary, but it's more for type safety and helping the developer understand what kind of error object they are getting. In a lot of our callback patterns, we return errors, but it's kind of willy-nilly in terms of how the error object is structured. Adding a type allows us to type check the error, and know that when we receive the error in a callback, we are guaranteed to have the property (ie `err.message`) that we are looking for.

Reply:
I definitively agree with you, the error interface is already defined, you don't need to redo it. See the interface is already like that:

```
interface Error {
    name: string;
    message: string;
    stack?: string;
}
```

Reply:
I removed the `errors.d.ts` import from `vm.ts` and viewing the definition of the resulting Error, shows that `Error~ is simply:
```
interface Error {
  stack?: string
}
```
See screenshot:
![Error definition](https://i.imgur.com/FbkByZH.png)

Is there something else I need to do to get the interface as you mentioned above? Do I need to import that from somewhere? Maybe DefinitelyTyped?

Reply:
Have updated the branch with this change so you can see it.

Reply:
See this commit: https://github.com/embark-framework/embark/pull/1234/commits/967cbe08e9cbf8610b2af6fe28ccd88353425612

Also on your screenshot on the right you can see 2 implementation, the correct one is in `lib.es5.d.ts`

![screenshot from 2019-01-13 12-12-14](https://user-images.githubusercontent.com/491074/51085161-3ca62780-172d-11e9-9f3d-67e747827fa7.png)

Feel free to dismiss my commit